bob/src/api/mod.rs

```diff
@@ -1143,6 +1152,175 @@ where
+    let pearl = group.detach_by_id(&partition_id).await.ok();
+    if let Some(holder) = pearl {
+        drop_directories(vec![holder], format!("id {} in vdisk {} on disk {}", partition_id, vdisk_id, disk_name)).await
+    } else {
+        let msg = format!(
+            "partition {} not found on vdisk {} on disk {} or it is active",
+            partition_id, vdisk_id, disk_name
+        );
+        Err(StatusExt::new(StatusCode::BAD_REQUEST, true, msg))
+    }
+}
+
+// DELETE /alien/disks/:disk_name/vdisks/:vdisk_id/partitions/:partition_id
+async fn alien_delete_partition_by_id<A>(
+    bob: Extension<BobServer<A>>,
+    AxumPath((node_name, vdisk_id, partition_id)): AxumPath<(String, u32, String)>,
+    creds: CredentialsHolder<A>,
+) -> Result<StatusExt, StatusExt>
+where
+    A: Authenticator,
+{
+    if !bob
+        .auth()
+        .check_credentials_rest(creds.into())?
+        .has_rest_write()
+    {
+        return Err(AuthError::PermissionDenied.into());
+    }
+    let group = find_alien_group_on_disk(&bob, &node_name, vdisk_id).await?;
+    let pearl = group.detach_by_id(&partition_id).await.ok();
+    if let Some(holder) = pearl {
+        drop_directories(vec![holder], format!("id {} in vdisk {} for node {}", partition_id, vdisk_id, node_name)).await
+    } else {
+        let msg = format!(
+            "alien partition {} not found on vdisk {} for node {} or it is active",
+            partition_id, vdisk_id, node_name
+        );
+        Err(StatusExt::new(StatusCode::BAD_REQUEST, true, msg))
+    }
+}
+
+async fn find_group_on_disk<A: Authenticator>(
+    bob: &BobServer<A>, 
+    disk_name: &str, 
+    vdisk_id: u32
+) -> Result<PearlGroup, StatusExt> {
+    let backend = bob.grinder().backend().inner();
+    let (dcs, _) = backend
+        .disk_controllers()
+        .ok_or_else(not_acceptable_backend)?;
+    find_disk_vdisk_group(dcs, disk_name, vdisk_id).await
+}
+
+async fn find_alien_group_on_disk<A: Authenticator>(
+    bob: &BobServer<A>, 
+    node_name: &str, 
+    vdisk_id: u32
+) -> Result<PearlGroup, StatusExt> {
+    let backend = bob.grinder().backend().inner();
+    let (_, adc) = backend
+        .disk_controllers()
+        .ok_or_else(not_acceptable_backend)?;
+    let groups = adc.groups();
+    let pearls = groups.read().await;
+    pearls.iter()
+        .find(|g| g.node_name() == node_name && g.vdisk_id() == vdisk_id)
+        .cloned()
+        .ok_or_else(|| { 
+            let msg = format!("Alien vdisk group for node {} vdisk {} not found", node_name, vdisk_id); 
+            StatusExt::new(StatusCode::NOT_FOUND, false, msg) 
+        })
+}
+
+async fn find_disk_vdisk_group(
+    dcs: &[std::sync::Arc<bob_backend::pearl::DiskController>], 
+    disk_name: &str, 
+    vdisk_id: u32
+) -> Result<PearlGroup, StatusExt> {
+    let needed_dc = dcs
+        .iter()
+        .find(|dc| dc.disk().name() == disk_name && dc.vdisks().iter().any(|&vd| vd == vdisk_id))
```

Comment:
Disk names are unique, so checking for `vdisk_id` is not required here. This check is done below when trying to get a vdisk from a disk controller.